clients/hadoop-to-venice-bridge/src/main/java/com/linkedin/venice/hadoop/input/kafka/VeniceKafkaInputReducer.java

```diff
@@ -27,7 +23,8 @@ public class VeniceKafkaInputReducer extends VeniceReducer {
   private static final RecordDeserializer<KafkaInputMapperValue> KAFKA_INPUT_MAPPER_VALUE_AVRO_SPECIFIC_DESERIALIZER =
       FastSerializerDeserializerFactory
           .getFastAvroSpecificDeserializer(KafkaInputMapperValue.SCHEMA$, KafkaInputMapperValue.class);
-  private ChunkAssembler chunkAssembler;
+  private ChunkAssembler chunkAssembler = null;
+  private MessageExtractor extractor = this::extractNonChunkedMessage;
```

Comment:
Sometimes this confuses me since we haven't explicitly specified extractNonChunkedMessage to proxy calls to `MessageExtractor::extract`. I remember seeing a similar pattern in the `KCSConstructor`.

Is this a feature in newer Java versions?

Reply:
It works in Java 8, so not that new... 🤷 ?